libs/utils/decorate.js

```diff
@@ -68,5 +53,61 @@ export function decorateBlockBg(block, node) {
 export function getBlockSize(el, defaultSize = 1) {
   const sizes = ['small', 'medium', 'large', 'xlarge'];
   if (defaultSize < 0 || defaultSize > sizes.length - 1) return null;
-  return sizes.find((size) => el.classList.contains(size)) || sizes[defaultSize];
+  return sizes.find((size) => el?.classList.contains(size)) || sizes[defaultSize];
+}
+
+function getDescendants(node, fragment = document.createDocumentFragment()) {
```

Comment:
1. Does it matter that nodes are added depth first?  (Order of child nodes gets changed)
2.  `appendChild` moves the nodes out of the DOM.  I think for your use case it's fine, but worth noting that this does modify the DOM as it traverses.

Reply:
1. I don't think so . I ran some more tests today and didn't see the order being affected for example
2. oh, I don't like that. This was actually also causing unexpected behaviour. I'm rewriting it now to stop the DOM being hit so often and fix that behaviour I was mentioning - thanks!